Avoid linking to edgeguides from source

zzak opened the pull request:
There is a [precedent](https://guides.rubyonrails.org/ruby_on_rails_guides_guidelines.html#linking-to-the-api) for linking to the stable version of our docs.

> Please don't link to edgeapi.rubyonrails.org manually.

I wonder if this should apply to all files? 🤔

p8 commented on actioncable/actioncable.gemspec:
```diff
@@ -30,7 +30,7 @@ Gem::Specification.new do |s|
 
   # NOTE: Please read our dependency guidelines before updating versions:
-  # https://edgeguides.rubyonrails.org/security.html#dependency-management-and-cves
+  # https://guides.rubyonrails.org/security.html#dependency-management-and-cves
```
These gemspec changes were added in: https://github.com/rails/rails/pull/34392 and https://github.com/rails/rails/pull/34388. So at the time linking to edge guides made sense as the regular guides didn't have this section yet.

p8 commented on CONTRIBUTING.md:
```diff
@@ -19,15 +19,15 @@
   * [**Generic template** for other issues](https://github.com/rails/rails/blob/main/guides/bug_report_templates/generic_main.rb)
 
-* For more detailed information on submitting a bug report and creating an issue, visit our [reporting guidelines](https://edgeguides.rubyonrails.org/contributing_to_ruby_on_rails.html#reporting-an-issue).
+* For more detailed information on submitting a bug report and creating an issue, visit our [reporting guidelines](https://guides.rubyonrails.org/contributing_to_ruby_on_rails.html#reporting-an-issue).
```
Maybe linking the contributing guides to edge still makes sense as these might change before getting a release.

skipkayhil reviewed (commented):
Related: #46482

Definitely :+1: on fixing the link in `activerecord/lib/arel.rb`, as well as the {framework}_overview links in READMEs because they show up here: https://api.rubyonrails.org/files/actioncable/README_md.html

zzak committed:
Use stable guides link for package READMEs

E.g.: These show up here:
https://api.rubyonrails.org/files/actioncable/README_md.html

zzak committed:
Refer to stable guide for ARel security
